pkg/virt-controller/services/virtiofs.go

```diff
@@ -99,26 +123,35 @@ func generateContainerFromVolume(volume *v1.Volume, image string) k8sv1.Containe
 	resources := resourcesForVirtioFSContainer(false, false)
 
 	socketPathArg := fmt.Sprintf("--socket-path=%s", VirtioFSSocketPath(volume.Name))
-	sourceArg := fmt.Sprintf("source=%s", virtioFSMountPoint(volume.Name))
-	args := []string{socketPathArg, "-o", sourceArg, "-o", "sandbox=chroot", "-o", "xattr", "-o", "xattrmap=:map::user.virtiofsd.:"}
+	sourceArg := fmt.Sprintf("--shared-dir=%s", virtioFSMountPoint(volume))
+	args := []string{socketPathArg, sourceArg, "--cache=auto", "--sandbox=chroot"}
+
+	if !isConfig(volume) {
+		args = append(args, "--xattr")
+	}
+
+	volumeMounts := []k8sv1.VolumeMount{
+		// This is required to pass socket to compute
+		{
+			Name:      virtiofs.VirtioFSContainers,
+			MountPath: virtiofs.VirtioFSContainersMountBaseDir,
+		},
+	}
+
+	if !isAutomount(volume) {
+		volumeMounts = append(volumeMounts, k8sv1.VolumeMount{
+			Name:      volume.Name,
+			MountPath: virtioFSMountPoint(volume),
```

Comment:
I think we don't need to complicate this. As is we have one container per volume. This means we can have one unified path to mount to. WDYT?

Reply:
Having one mount point will not work for `ServiceAccount` since it's marked as automount, so the virtiofs container doesn't have the Volume definition. So, I still need to check `if !isAutoMount()` (*), the only difference will be in `virtioFSMountPoint()` returning "/volume_name" for all volumes but `ServiceAccount`, so I think it's preferable keeping the defined mount point just for consistency.